app/ports/views.py

```diff
@@ -151,18 +151,28 @@ def portdetail_build_information(request):
 
 
 def portdetail_stats(request):
+    days = request.GET.get('days', 30)
+    days_ago = request.GET.get('days_ago', 0)
+    try:
+        days = int(days)
+        days_ago = int(days_ago)
+    except ValueError:
+        return HttpResponse("'days' and 'days-ago' must be integer.")
     port_name = request.GET.get('port_name')
     port = Port.objects.get(name__iexact=port_name)
 
+    end_date = datetime.datetime.now(tz=datetime.timezone.utc) - datetime.timedelta(days_ago)
+    start_date = end_date - datetime.timedelta(days=days)
```

Comment:
We have (or will have) the same functionality on `/statistics/`. It makes sense to write a function that will return two datetime objects from the given (`days`, `days_ago`).